core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -230,20 +221,58 @@ oppia.factory('TopicUpdateService', [
           // Get the current change list.
           var currentChangeList = UndoRedoService.getChangeList();
           var indicesToDelete = [];
+          for (var i = 0; i < currentChangeList.length; i++) {
```

Comment:
This is complex, so add some comments here explaining clearly _why_ you're doing what you're doing.

Reply:
Done, added comments.